### PR TITLE
Refactor test_generate_hashes_without_interfering_with_each_other.

### LINKED ISSUE #1002: Refactor test_generate_hashes_without_interfering_with_each_other
This test checks a very special case with `cffi` and `maptolib`, see 877241d14d84a97801e98f4be473e5c63de5e717. Could we leave it as is and refactor it in the following PR?

_Originally posted by @atugushev in https://github.com/jazzband/pip-tools/pull/998#discussion_r349934456_

---

I think we should describe in detail what exactly this test checks, since it's not obvious what could go wrong with `PyPIRepository.get_hashes()` and add links to #512 and #514 issues (I've spent too much time to track these issues). For example:
```python
    """
    The PyPIRepository._get_file_hash() used to call unpack_url(),
    when generating the hash. Unpacking both packages to the same directory
    will then fail. E.g. matplotlib-2.0.2.tar.gz has a directory named LICENSE,
    but many other packages have a file named LICENSE.

    See GH-512 and GH-544.
    """"
```

and assert the result of `get_hashes()`.

### PR DESCRIPTION
**Pull request by hramezani**
Fixes #1002.
Thanks @atugushev for description.

**Changelog-friendly one-liner**: Refactor `test_generate_hashes_without_interfering_with_each_other`.

##### Contributor checklist

- [x] Provided the tests for the changes.
- [x] Gave a clear one-line description in the PR (that the maintainers can add to CHANGELOG.md on release).
- [ ] Assign the PR to an existing or new milestone for the target version (following [Semantic Versioning](https://blog.versioneye.com/2014/01/16/semantic-versioning/)).

### CODE REVIEW

**Comment by hramezani** [comment]
@atugushev , Should I use `with pypi_repository.allow_all_wheels()` for this test as well?

**Comment by hramezani** [comment]
@atugushev Thanks for your suggestion.

File: tests/test_repository_pypi.py [hunk 1]
```diff
@@ -61,8 +61,35 @@ def test_generate_hashes_all_platforms(from_line, pypi_repository):
 def test_generate_hashes_without_interfering_with_each_other(
```
Comment (atugushev) [inline tests/test_repository_pypi.py]: Probably we should also rename the test name.

Comment (hramezani) [reply tests/test_repository_pypi.py]: what about `test_get_file_hash_without_interfering_with_each_other`?

Comment (atugushev) [reply tests/test_repository_pypi.py]: yeah, looks good 👍

**Pull request merged**